featuretools/tests/primitive_tests/test_transform_features.py

```diff
@@ -699,7 +701,6 @@ def test_boolean_multiply(boolean_mult_es):
     to_test = [
         ("numeric", "numeric"),
         ("numeric", "bool"),
-        ("bool", "numeric"),
```

Comment:
why was this case removed?

Reply:
We can't swap inputs with the new `MultiplyBooleanNumeric` primitive like we could with the previous `MultiplyNumeric` primitive. The first input needs to be the numeric column and the second the boolean column.

Or if you prefer, we could add logic in the primitive to determine which is which based on dtype and act accordingly?

Reply:
what if we add logic in the `__mul__` override to pass in [numeric, bool] in that order instead of always doing [self, other]?

Reply:
Updated the primitive logic and inputs to handle this case since updating `__mul__` could result in two features with the same name.